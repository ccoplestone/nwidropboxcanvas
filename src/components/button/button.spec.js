import React from 'react';
import { shallow } from 'enzyme';

import { Link } from 'react-router-dom';

import Button from './index';

describe('Button', () => {
    it('should render', () => {
        expect(shallow(<Button>Lorem</Button>)).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(shallow(<Button>Lorem</Button>).text()).toBe('Lorem');
    });
    it('should have a default onClick', () => {
        expect(Button.defaultProps.onClick()).toBe(null);
    });
    it('should be disabled', () => {
        const component = shallow(<Button isDisabled>Lorem</Button>);
        expect(component.prop('disabled')).toBe(true);
    });
    it('should render link', () => {
        const component = shallow(<Button to="lorem">Lorem</Button>);
        expect(component.type()).toBe(Link);
        expect(component.prop('to')).toBe('lorem');
    });
    it('should handle click', () => {
        let number = 0;
        const component = shallow(<Button onClick={() => { number += 1; }}>Lorem</Button>);

        expect(number).toBe(0);
        component.simulate('click');
        expect(number).toBe(1);
    });
});
