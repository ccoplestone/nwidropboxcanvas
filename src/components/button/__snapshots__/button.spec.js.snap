// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    isDisabled={false}
    onClick={[Function]}
    to={null}
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
